cpp/src/arrow/array/builder_primitive.h

```diff
@@ -43,6 +43,14 @@ class ARROW_EXPORT NullBuilder : public ArrayBuilder {
   /// \brief Append a single null element
   Status AppendNull() final { return AppendNulls(1); }
 
+  Status AppendEmptyValues(int64_t length) final {
+    if (length < 0) return Status::Invalid("length must be positive");
+    length_ += length;
```

Comment:
For the null type, `AppendEmptyValues` should simply be the same as `AppendNulls` (the only possible value is null).